### ISSUE TITLE
Update python func to new programming model

### ISSUE THREAD

**Issue by jongio** [Azure/azure-dev#1232]
This new python func programming model was released.
https://github.com/Azure/azure-functions-python-worker/wiki/V2-Programming-Model-for-Azure-Functions-using-Python

We should consider upgrading our python func project to it.

**Comment by pamelafox** [comment]
FYI, for your reference, here's a PR where I updated my own azd FastAPI template:
https://github.com/pamelafox/fastapi-azure-function-apim/pull/19/files

**Comment by jongio** [comment]
@pamelafox - No worries if you don't have time, but would love a contrib on it to ours...

**Comment by pamelafox** [comment]
@jongio Yeah I figured I was setting myself up for that, I'll put on my TODO list. :)
